src/services/anchor-service.ts

```diff
@@ -78,74 +76,113 @@ class IpfsLeafCompare implements CompareFunction<Candidate> {
 /**
  * Anchors CIDs to blockchain
  */
-export default class AnchorService implements Contextual {
-  private ipfsMerge: IpfsMerge;
-  private ipfsCompare: IpfsLeafCompare;
-
-  private requestService: RequestService;
-  private ceramicService: CeramicService;
-  private blockchainService: BlockchainService;
-
-  private anchorRepository: AnchorRepository;
-
-  /**
-   * Initialize the service
-   */
-  public async init(): Promise<void> {
-    this.ipfsMerge = new IpfsMerge(this.ceramicService.ipfs);
+@singleton()
+export default class AnchorService {
+  private readonly ipfsMerge: IpfsMerge;
+  private readonly ipfsCompare: IpfsLeafCompare;
+
+  constructor(
+    @inject('blockchainService') private blockchainService?: BlockchainService,
+    @inject('ipfsService') private ipfsService?: IpfsService,
+    @inject('requestService') private requestService?: RequestService,
+    @inject('ceramicService') private ceramicService?: CeramicService,
+    @inject('anchorRepository') private anchorRepository?: AnchorRepository) {
+
+    this.ipfsMerge = new IpfsMerge(this.ipfsService);
     this.ipfsCompare = new IpfsLeafCompare();
   }
 
-  /**
-   * Sets dependencies
-   * @param context - Application context
-   */
-  setContext(context: Context): void {
-    this.blockchainService = context.getSelectedBlockchainService();
-
-    this.ceramicService = context.lookup('CeramicService');
-    this.requestService = context.lookup('RequestService');
-    this.anchorRepository = context.lookup('AnchorRepository')
-  }
-
   /**
    * Finds anchor by request
    * @param request - Request instance
    */
   public async findByRequest(request: Request): Promise<Anchor> {
-    return this.anchorRepository.findByRequest(request)
+    return this.anchorRepository.findByRequest(request);
   }
 
   /**
    * Creates anchors for client requests
    */
   public async anchorRequests(): Promise<void> {
-    const reqs: Request[] = await this.requestService.findNextToProcess();
-    if (reqs.length === 0) {
-      logger.Info('No pending CID requests found. Skipping anchor.');
+    logger.Imp('Anchoring pending requests...');
+
+    let requests: Request[] = await this.requestService.findNextToProcess();
+    if (requests.length === 0) {
+      logger.Info("No pending CID requests found. Skipping anchor.");
```

Comment:
We should definitely use more `debug` instead of `Info` or `Imp`. The `@overnightjs` logger implementation has some limitation with logging levels but that will change soon.